File: RxTools-library/src/main/java/com/vondear/rxtools/view/progress/sprite/RingSprite.java
package com.vondear.rxtools.view.progress.sprite;

import android.animation.ValueAnimator;
import android.graphics.Canvas;
import android.graphics.Paint;

/**
 * Created by ybq.
 */
public class RingSprite extends ShapeSprite {


    @Override
    public void drawShape(Canvas canvas, Paint paint) {
        if (getDrawBounds() != null) {
            paint.setStyle(Paint.Style.STROKE);
            int radius = Math.min(getDrawBounds().width(), getDrawBounds().height()) / 2;
            paint.setStrokeWidth(radius / 12);
            canvas.drawCircle(getDrawBounds().centerX(),
                    getDrawBounds().centerY(),
                    radius, paint);
        }
    }

    @Override
    public ValueAnimator onCreateAnimation() {
        return null;
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/progressing/sprite/RingSprite.kt
package com.tamsiree.rxui.view.progressing.sprite

import android.animation.ValueAnimator
import android.graphics.Canvas
import android.graphics.Paint

/**
 * @author tamsiree
 */
open class RingSprite : ShapeSprite() {
    override fun drawShape(canvas: Canvas?, paint: Paint?) {
        if (drawBounds != null) {
            paint!!.style = Paint.Style.STROKE
            val radius = Math.min(drawBounds.width(), drawBounds.height()) / 2
            paint.strokeWidth = radius / 12.toFloat()
            canvas!!.drawCircle(drawBounds.centerX().toFloat(),
                    drawBounds.centerY().toFloat(),
                    radius.toFloat(), paint)
        }
    }

    override fun onCreateAnimation(): ValueAnimator? {
        return null
    }
}